edk2toollib/uefi/edk2/parsers/base_parser.py

```diff
@@ -193,6 +194,21 @@ def PopConditional(self):
             self.Logger.critical("Tried to pop an empty conditional stack.  Line Number %d" % self.CurrentLine)
             return self.ConditionalStack.pop()  # this should cause a crash but will give trace.
 
+    def _FindReplacementForToken(self, token):
+
+        v = self.LocalVars.get(token)
+
+        if(v is None):
+            v = self.InputVars.get(token)
+
+        if(v is None):
+            v = self.MacroNotDefinedValue
+
+        if(v.upper() == "TRUE" or v.upper == "FALSE"):
```

Comment:
v.upper() == "FALSE" instead of v.upper ==

Reply:
My guess is that this condition isn't being tested if we can short circuit it

Reply:
Do we want to check if token is TRUE or FALSE, or just look at the replacement value?